lib/keila_web/views/sender_view.ex

```diff
@@ -1,3 +1,37 @@
 defmodule KeilaWeb.SenderView do
   use KeilaWeb, :view
+
+  def sender_adapters do
+    Keila.Mailings.SenderAdapters.adapter_names()
+  end
+
+  def sender_adapter_name("smtp"), do: "SMTP"
+  def sender_adapter_name("ses"), do: "SES"
+  def sender_adapter_name("sendgrid"), do: "Sendgrid"
+  def sender_adapter_name("mailgun"), do: "Mailgun"
+  # TODO: Replace me. Currently needed to support test sender adapter.
+  if Mix.env() == :test do
+    def sender_adapter_name("test"), do: "Test"
+  end
+
+  def render_sender_adapter_form(form, "smtp") do
+    render("_smtp_config.html", form: form)
+  end
+
+  def render_sender_adapter_form(form, "ses") do
+    render("_ses_config.html", form: form)
+  end
+
+  def render_sender_adapter_form(form, "sendgrid") do
+    render("_sendgrid_config.html", form: form)
+  end
+
+  def render_sender_adapter_form(form, "mailgun") do
+    render("_mailgun_config.html", form: form)
+  end
+
+  # TODO: Replace me. Currently needed to support test sender adapter.
```

Comment:
I'm not 100% sure.

I don't like having `if Mix.env` in the production code.

Since we don't want to leak KeilaWeb concerns into the sender adapters we have the following options:

- Add a module somewhere in KeilaWeb for each sender adapter and let that module have a `sender_adapter` method that returns the sender adapter. Then use KeilaWeb to configure a list of those modules instead of a list of sender adapters.

- Remove the need for the `test` sender adapter.

- Keep this as it is.

Now, for a hex library the furst approach would make the most sense so that it is easy to extend the lib.
But here we might just get away with keeping it as it is. We can still change that later if we need to.

Reply:
How about adding a configuration `:keila, :env, :prod/:dev/:test`? Then this kind of code would be independent from how the environment is actually configured.

Reply:
I think that's just hiding the real problem. It's basically code that is only there to support testing.

I think it's ok to keep it like that as long as it stays minimal. We can make the statement that the backend part of sender adapters is modular while the frontend currently isn't. The fact that we can at least guard with `Mix.env` is actually good since that at least means that we don't compile test-only code when running in production.

Reply:
Okay, sounds good to me :+1: